docs/components/plugins/JsonSchema/DefinitionType.tsx

```diff
@@ -0,0 +1,79 @@
+import { DefinitionRenderProps } from './Definition';
+import { resolveRef } from './utils/reference';
+import { Schema, SchemaProperty } from './utils/types';
+
+import { CODE } from '~/ui/components/Text';
+
+export function DefinitionType(props: DefinitionRenderProps) {
+  const type = getPropertyType(props.schema, props.definition);
+  return !type ? null : <CODE css={{ whiteSpace: 'nowrap' }}>{type}</CODE>;
+}
+
+/**
+ * Get the type of a single JSON Schema property to render.
+ * For primitive types, the text is returned.
+ * For complex types, such as array, the `items` types are resolved.
+ */
+export function getPropertyType(
```

Comment:
This one needs some work. We might need to link to another header whenever we encounter a referred object type. To keep things readable, but I'm slightly struggling with that.